exp/runtime/hooks/api/v1alpha1/common_types.go

```diff
@@ -70,3 +79,21 @@ const (
 	// ResponseStatusFailure represents a failure response.
 	ResponseStatusFailure ResponseStatus = "Failure"
 )
+
+// CommonRetryResponse is the data structure which contains all
+// common retry fields.
+type CommonRetryResponse struct {
```

Comment:
Q: should this a CommonResponse + RetryAfterSeconds, so we can satisfy the interface with a single object?

Reply:
What seems strange to me is that we have to add both CommonRespose + CommonRetryResponse when they all are used together; also I'm not worried about having more struct, we don't have a concrete use case for that now.

Instead, I think it is fine to keep TypeMeta separated, it is a common pattern in Kubernetes API objects.

Reply:
Another point, is there a use case where we expect to have CommonRetryResponse without CommonResponse? If not, then we should embed CommonResponse into CommonRetryResponse

Reply:
Done